--- temp/phononwidget.cpp	2014-02-25 19:41:45.000000000 +0100
+++ phononwidget.cpp	2014-03-20 11:11:53.801971476 +0100
@@ -30,6 +30,7 @@
 #include <QHBoxLayout>
 #include <QShowEvent>
 #include <QToolButton>
+#include <QCheckBox>
 
 #include <KDialog>
 #include <KIcon>
@@ -66,6 +67,7 @@
     m_stopButton(0),
     m_topLayout(0),
     m_media(0),
+    m_playAut(0),
     m_seekSlider(0),
     m_audioOutput(0),
     m_videoPlayer(0)
@@ -77,6 +79,9 @@
     if (m_url != url) {
         stop(); // emits playingStopped() signal
         m_url = url;
+        if (m_playAut ->isChecked()){
+            play();
+        }
     }
 }
 
@@ -116,10 +121,12 @@
         m_playButton = new QToolButton(this);
         m_stopButton = new QToolButton(this);
         m_seekSlider = new Phonon::SeekSlider(this);
+        m_playAut = new QCheckBox(this);
 
         controlsLayout->addWidget(m_playButton);
         controlsLayout->addWidget(m_stopButton);
         controlsLayout->addWidget(m_seekSlider);
+        controlsLayout->addWidget(m_playAut);
 
         m_topLayout->addLayout(controlsLayout);
 
@@ -140,6 +147,9 @@
         connect(m_stopButton, SIGNAL(clicked()), this, SLOT(stop()));
 
         m_seekSlider->setIconVisible(false);
+        
+        m_playAut->setToolTip(i18n("play automatically"));
+        m_playAut->setIconSize(buttonSize);
 
         // Creating an audio player or video player instance might take up to
         // 2 seconds when doing it the first time. To prevent that the user
